security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -96,21 +91,18 @@ func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) (res []s
 	}
 	// TODO(hzxuzhonghu): notify caclient rebuilding only when root cert is updated.
 	// It can happen when the istiod dns certs is resigned after root cert is updated,
-	// in this case, the ca grpc client can not automiatically connect to istiod after the underlying network connection closed.
+	// in this case, the ca grpc client can not automatically connect to istiod after the underlying network connection closed.
 	// Becase that the grpc client still use the old tls configuration to reconnect to istiod.
 	// So here we need to rebuild the caClient in order to use the new root cert.
 	defer func() {
-		if err != nil && strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
```

Comment:
Why is the `strings.Contains()` check no longer relevant?

Reply:
Because the UT allows present cert later, it is:
 first, the connection has been established with no certs provided
then secondly a CSRSign request is sent, will return authn failure. 
Thirdly, with the certs provided, a new CSRSign can succeed